### ISSUE TITLE
Order is wrong

### ISSUE THREAD

**Issue by symeonmattes** [ShankyTiwari/ng-material-multilevel-menu#69]
Hi,

I have put the items in ng-material-multilevel-menu as suggested but the items in the menu are shown in almost alphabetical order instead of the order I have put them. For instance:

`<ng-material-multilevel-menu [configuration]='config' [items]='menuItems' (selectedItem)="selectedItem($event)"></ng-material-multilevel-menu>`

```
menuItems = [
  {
    "label": "Fashion Feed",
    "link": "/feed",
    "imageIcon": "/assets/icons/feed-menu-icon.svg"
  },
  {
    "label": "Search",
    "link": "/search",
    "imageIcon": "/assets/icons/search-menu-icon.svg"
  },
  {
    "label": "Categories",
    "imageIcon": "/assets/icons/category-menu-icon.svg",
    "items": [
      {
        "label": "Fall/Winter 19",
        "link": "/feed?category=1&title=Fall%2FWinter%2019"
      },
      {
        "label": "Activewear",
        "link": "/feed?category=2&title=Activewear"
      },
      {
        "label": "Outerwear",
        "link": "/feed?category=3&title=Outerwear"
      },
      {
        "label": "Skirts",
        "link": "/feed?category=4&title=Skirts"
      },
      {
        "label": "Sleepers",
        "link": "/feed?category=5&title=Sleepers"
      },
      {
        "label": "Sweaters",
        "link": "/feed?category=6&title=Sweaters"
      },
      {
        "label": "Sweatshirts",
        "link": "/feed?category=7&title=Sweatshirts"
      },
      {
        "label": "T-shirts",
        "link": "/feed?category=8&title=T-shirts"
      },
      {
        "label": "Tanks & Camisoles",
        "link": "/feed?category=9&title=Tanks%20%26%20Camisoles"
      },
      {
        "label": "Tops",
        "link": "/feed?category=10&title=Tops"
      },
      {
        "label": "Beachwear",
        "link": "/feed?category=11&title=Beachwear"
      },
      {
        "label": "Blazers",
        "link": "/feed?category=12&title=Blazers"
      },
      {
        "label": "Bodysuits",
        "link": "/feed?category=13&title=Bodysuits"
      },
      {
        "label": "Bridal",
        "link": "/feed?category=14&title=Bridal"
      },
      {
        "label": "Denim",
        "link": "/feed?category=15&title=Denim"
      },
      {
        "label": "Dresses",
        "link": "/feed?category=16&title=Dresses"
      },
      {
        "label": "Jumpsuits & Rompers",
        "link": "/feed?category=17&title=Jumpsuits%20%26%20Rompers"
      },
      {
        "label": "Kimonos & Caftans",
        "link": "/feed?category=18&title=Kimonos%20%26%20Caftans"
      },
      {
        "label": "Leggings",
        "link": "/feed?category=19&title=Leggings"
      },
      {
        "label": "Lingerie",
        "link": "/feed?category=20&title=Lingerie"
      },
      {
        "label": "Overalls",
        "link": "/feed?category=21&title=Overalls"
      },
      {
        "label": "Pants",
        "link": "/feed?category=22&title=Pants"
      },
      {
        "label": "Shirts",
        "link": "/feed?category=23&title=Shirts"
      },
      {
        "label": "Shorts",
        "link": "/feed?category=24&title=Shorts"
      }
    ]
  },
  {
    "label": "Saved Items",
    "link": "/favorites",
    "imageIcon": "/assets/icons/saveditems-menu-icon.svg"
  }
]
```
and the way they appear is:

```
Fashion Feed
Search
Categories
- Fall/Winter 19
- Activewear
- Beachwear
- Blazers
- Bodysuits
- Bridal
- Denim
- Dresses
- Jumpsuits & Rompers
- Kimonos & Caftans
- Leggings
- Lingerie
- Outerwear
- Overalls
- Pants
- Shirts
- Shorts
- Skirts
- Sleepers
- Sweaters
- Sweatshirts
- T-shirts
- Tanks & Camisoles
- Tops
Saved Items
```

I've tried to check how you have implemented it but I didn't manage to figure it out. Do you have any ordering or what I should check?

Thanks

**Comment by cityinspain** [comment]
Hi,

I've also encountered this. I'm fairly certain that it occurs when a single list item has too many children, although I haven't been able to figure out why. I'm fairly certain that you can work around this by separating "Categories" into a few sublists. Would you mind sharing your code, so I can look into it?

**Comment by knpaing** [comment]
Hi,

I'm having the same issue too though I don't have long children on menus. Here is my menu:

 appitems = [
    {
      label: 'Dashboard',
      faIcon: 'fas fa-th',
      link: 'dashboard',
      selected: true
    },
    {
      label: 'Sales',
      icon: 'attach_money',
      link: 'sales'
    },
    {
      label: 'Purchases',
      icon: 'money_off',
      link: 'purchase'
    },
    {
      label: 'Expenses',
      icon: 'credit_card',
      link: 'expenses'
    },
    {
      label: 'Payables',
      faIcon: 'fas fa-money-bill-wave',
      items: [
        { label: 'Summary', icon: 'list', link: 'payable' },
        { label: 'Pay', icon: 'trending_down', link: 'payable_pay' }
      ]
    },
    {
      label: 'Receivables',
      faIcon: 'fas fa-file-invoice-dollar',
      items: [
        { label: 'Summary', icon: 'list', link: 'receivable' },
        { label: 'Receive', icon: 'trending_up', link: 'receivable_receipt' }
      ]
    },
    {
      label: 'Loans',
      faIcon: 'fas fa-money-check-alt',
      items: [
        { label: 'Registeration', icon: 'list', link: 'loan' },
        { label: 'Pay', icon: 'trending_down', link: 'loan_payment' }
      ]
    },
    {
      label: 'Inventory Control',
      faIcon: 'fas fa-cubes',
      link: 'inventory'
    },
    {
      label: 'Profit & Lost',
      icon: 'assessment',
      link: 'profitnlost'
    },
    {
      label: 'Balance Sheet',
      faIcon: 'fas fa-chart-bar',
      link: 'balancesheet'
    },
    {
      label: 'Closing',
      faIcon: 'fas fa-calendar-alt',
      link: 'closing'
    },
    {
      label: 'Customers',
      faIcon: 'fas fa-address-book',
      link: 'customer'
    },
    {
      label: 'Suppliers',
      faIcon: 'fas fa-truck',
      link: 'supplier'
    },
    {
      label: 'Finances',
      faIcon: 'fas fa-user-tie',
      link: 'finances'
    },
    {
      label: 'Capital',
      faIcon: 'fas fa-cash-register',
      link: 'capital'
    },
    {
      label: 'Setup',
      icon: 'star_rate',
      items: [
        { label: 'Users', faIcon: 'fas fa-users', link: 'user' },
        { label: 'Opening Balance', faIcon: 'fas fa-calendar-check', link: 'opening' },
        { label: 'Expense Category', faIcon: 'fas fa-th-large', link: 'expensecategory' }
      ]
    }
  ];

And it appears something like appeared in attachment image.
![Screen Shot 2019-09-05 at 6 19 56 PM](https://user-images.githubusercontent.com/12625702/64339359-d9de0500-d009-11e9-9e6f-dd23181b29d5.png)


Thanks.

**Comment by symeonmattes** [comment]
> Hi,
> 
> I've also encountered this. I'm fairly certain that it occurs when a single list item has too many children, although I haven't been able to figure out why. I'm fairly certain that you can work around this by separating "Categories" into a few sublists. Would you mind sharing your code, so I can make sure there's nothing else wrong?

Hi...I haven't checked what you're saying. It doesn't make sense to change the order regardless of whether the list is short or long. Unfortunately it's not an opensource project so I can't give it to you. If you make in angular a page and you put what I have said when I started the issue including:

```
  public config: any = {
    paddingAtStart: true,
    classname: 'menu-items',
    listBackgroundColor: '#fff',
    fontColor: '#000',
    backgroundColor: '#fff',
    selectedListFontColor: '#000',
    highlightOnSelect: false
  };
```
you will have the same result

**Comment by knpaing** [comment]
Hi,

Thank you for your reply. I just tested and when the menu list goes over 10 items, it goes weird with the orders. I mean not including the children in my case. Can you please check with 10 menu items and then add another one? 

Thank you for the great work!

**Comment by ShankyTiwari** [comment]
Okay, I'll check it ASAP and sorry for replying late.
Thanks.

**Comment by sauravg666** [comment]
getting this error in browser's console

NgMaterialMultilevelMenuComponent.html:1 ERROR TypeError: Cannot read property 'length' of undefined
    at Object.eval [as updateDirectives] (NgMaterialMultilevelMenuComponent.html:1)

**Comment by tonywang160** [comment]
Hi,  I have the same order issue

`
config: Configuration = {
    paddingAtStart: true,           // boolean => [optional] If you don't want padding at the start of the list item, then you can give false. The default value will be true.
    interfaceWithRoute: true,       // boolean => [required] only if you want to use Angular Routing with this menu.
    highlightOnSelect: true,        // boolean => [optional] If you want to highlight the clicked item in the list, then you can do that by making it true. The default value will be false.
    collapseOnSelect: true,         // boolean; => [optional] You have the option to collapse another parent when clicked on the current parent. The default value will be false.
    rtlLayout: false,               // boolean; => [optional] whether display is Right To Left. The default value will be false.
    classname: 'menu',              // string; => [optional] You can give your own custom class name in order to modify the list appearance.
    listBackgroundColor: 'white',   // string; => [optional] You can apply custom color to the background of the list.
    fontColor: 'black',             // string; => [optional] Changes the color of Text and icons inside the list.
    backgroundColor: 'white',       // string; => [optional] This will change the background color list container.
    selectedListFontColor: environment.cssVar.primaryColor,   // string; => [optional] This will change the font color of selected list item.
  }
`

`
{
      label: '育達高級中學',
      link: '/main',
      // imageIcon: '/assets/batman.jpg',

      externalRedirect: false,
      items: [
        {
          label: '教學單位',
          icon: 'school',
          items: [
            {
              label: '招生科別',
              link: '/main/pages/depts',
              icon: 'keyboard_arrow_right',
            },
            {
              label: '高中部(普通科)',
              icon: 'keyboard_arrow_right',
              items: [
                {
                  label: '普通科菁英班',
                  link: '/main/dept/普通科菁英班',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '普通科資優班',
                  link: '/main/dept/普通科資優班',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '普通科',
                  icon: 'keyboard_arrow_right',
                  link: '/main/dept/普學',
                },
              ]
            },
            {
              label: '高職部',
              icon: 'keyboard_arrow_right',
              items: [
                {
                  label: '高職部菁英班',
                  link: '/main/dept/高職部菁英班',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '特色招生班',
                  link: '/main/dept/高職部特色招生班',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '會計事務科',
                  link: '/main/dept/職會',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '商業經營科',
                  link: '/main/dept/職商',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '國際貿易科',
                  link: '/main/dept/職貿',
                  icon: 'keyboard_arrow_right',
                },
                {

                  label: '廣告設計科',
                  link: '/main/dept/職廣',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '多媒體設計科',
                  link: '/main/dept/職多',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '資料處理科',
                  link: '/main/dept/職資',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '幼兒保育科',
                  link: '/main/dept/職幼',
                  icon: 'keyboard_arrow_right',
                },
                {
                  label: '應用外語科(英文組)',
                  link: '/main/dept/職英',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '應用外語科(日文組)',
                  link: '/main/dept/職日',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '時尚造型科',
                  link: '/main/dept/職美',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '餐飲管理科',
                  link: '/main/dept/職餐',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '觀光事業科',
                  link: '/main/dept/職觀',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                {
                  label: '資訊科',
                  link: '/main/dept/職訊',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
              ]
            },

            {
              label: '實用技能學程',
              icon: 'keyboard_arrow_right',
              items: [
                {
                  label: '餐飲技術科',
                  link: '/main/dept/實餐',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                },
                // {
                //   label: '廣告技術科',
                //   link: '/main/dept/實廣',
                //   icon: 'keyboard_arrow_right',
                //   onSelected: function () {
                //     console.log('Item 1.2.2.1');
                //   }
                // },
                {
                  label: '美容造型科',
                  link: '/main/dept/實美',
                  icon: 'keyboard_arrow_right',
                  onSelected: function () {
                    console.log('Item 1.2.2.1');
                  }
                }
              ]
            },
            {
              label: '進修部',
              link: '/main/dept/進修學校教學單位',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
          ]
        },
        {
          label: '行政單位',
          icon: 'work',
          items: [
            {
              label: '校長室',
              link: '/main/office/校長室',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '教務處',
              link: '/main/office/教務處',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '學務處',
              link: '/main/office/學務處',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '進修部',
              link: '/main/office/進修學校',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '總務處',
              link: '/main/office/總務處',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '人事室',
              link: '/main/office/人事室',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '會計室',
              link: '/main/office/會計室',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '圖書館',
              link: '/main/office/圖書館',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '心理輔導室',
              link: '/main/office/心理輔導室',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '資訊中心',
              link: '/main/office/資訊中心',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
          ]
        },
        {
          label: '關於育達',
          icon: 'info',
          items: [
            {
              label: '本校簡介',
              link: '/main/bbp/article/桃育本校簡介',
              icon: 'keyboard_arrow_right',
            },
            {
              label: '育達頻道',
              link: '/main/youtube',
              icon: 'keyboard_arrow_right',
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '升學英雄榜',
              link: '/main/college-result/board/197',   // 197 為 桃育升學英雄榜
              icon: 'keyboard_arrow_right',
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '學生社團',
              link: '/main/society',
              icon: 'keyboard_arrow_right',
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '育達週刊',
              link: '/main/bbp/board/桃育育達週刊',
              icon: 'keyboard_arrow_right',
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '活動照片',
              link: '/main/bbp/board/桃育活動照片',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '媒體報導',
              link: '/main/bbp/board/桃育媒體報導',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '外賓參訪紀錄',
              link: '/main/bbp/board/桃育外賓參訪紀錄',
              icon: 'keyboard_arrow_right',
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '交通指南',
              link: '/main/bbp/article/桃育交通指南',
              icon: 'keyboard_arrow_right',
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '推廣教育中心',
              link: '/epc',
              icon: 'keyboard_arrow_right',
              hidden: true,
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
            {
              label: '校友辦公室',
              link: '/alu',
              icon: 'keyboard_arrow_right',
              hidden: true,
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '桃園市平鎮區教育會',
              link: '/PEA',
              icon: 'keyboard_arrow_right',
              hidden: true,
              onSelected: function () {
                console.log('Item 1.2.2.1');
              }
            },
            {
              label: '台北育達高職',
              link: 'http://www.yudah.tp.edu.tw/',
              externalRedirect: true,
              hidden: true,
              // target: '_blank',
              icon: 'keyboard_arrow_right',
            },
            {
              label: '育達科技大學',
              link: 'http://www.ydu.edu.tw/',
              externalRedirect: true,
              icon: 'keyboard_arrow_right',
              hidden: true,
              // onSelected: function () {
              //   console.log('Item 1.2.2.1');
              // }
            },
          ]
        },
        {
          label: '新生專區',
          link: '/main/pages/new-students',
          icon: 'person_add',
        },
        {
          label: '學生及家長',
          link: '/main/pages/students',
          icon: 'person',
        },
        {
          label: '校友',
          link: '/main/pages/alumni',
          icon: 'person',
          hidden: true,
          onSelected: function () {
            console.log('Item 3');
          }
        },
        {
          label: '教職員',
          link: '/main/pages/employee',
          icon: 'person',
          hidden: true,
        },
        {
          label: '常用連結',
          link: '/main/bbp/board/桃育_常用連結?boardback=0',
          icon: 'keyboard_arrow_right',
          hidden: true,
          items: [
            {
              label: 'test',
              link: '/test',
              icon: 'keyboard_arrow_right',
              hidden: true,
            },
          ],
        },
        {
          label: 'test',
          link: '/test',
          icon: 'keyboard_arrow_right',
          hidden: true
        },
      ]
    }
`

**Comment by ShankyTiwari** [comment]
Fixed in  [4.11.0](https://www.npmjs.com/package/ng-material-multilevel-menu/v/4.11.0). 
Here is a [PR](https://github.com/ShankyTiwari/ng-material-multilevel-menu/pull/78).

**Comment by suraj740** [comment]
Hi @ShankyTiwari , how to order menu items in aplhabetical order.
